### ISSUE TITLE
Virtual Line times at Europa-Park

### ISSUE THREAD

**Issue by HookDonn** [cubehouse/themeparks#294]
Hello,
Would it be possible to get the Virtual Line times from the attractions at Europa-Park?
I know it's possible but I don't know how... :(

Thanks you, Hook.

**Comment by LouisForaux** [comment]
Hi

In the current API, there is no way for getting virtual line times. 

Maybe in an other URL but for now, I can't reach them

**Comment by HookDonn** [comment]
Yes I think it's via another URL but I don't know where to find it or how to do it with the application and certain software

**Comment by HoffiF** [comment]
I have been working on the new API of Europa-Park, which also displays the VirtualLine and the waterpark Rulantica, which can be found at the following URLs:

JWT Authorization: https://tickets.mackinternational.de/api/v1/login_check"
Per Body must be transferred here:
{"password": "", "username": ""}

Opening hours Europa-Park: https://tickets.mackinternational.de/api/v1/opentime/europapark/de
Opening hours Europa-Park Hotel: https://tickets.mackinternational.de/api/v1/hoteltime/europapark
Opening hours Rulantica: https://tickets.mackinternational.de/api/v1/hoteltime/rulantica
Opening hours Rulantica Hotel: https://tickets.mackinternational.de/api/v1/opentime/rulantica/de

Waiting times Europa-Park & Rulantica: https://tickets.mackinternational.de/api/v1/waitingtimes

However, I currently can't get the username and password to pick up the JWT Authorization Bearer Token.

**Comment by HoffiF** [comment]
Small feedback, the new version of 8.0.0 seems to use Certificate Transparency to prevent a man-in-the-middle attack. I can't get around the SSL pinning with it currently. The access data of the older version 7.2.0. where you could still intercept the data is disabled. Maybe someone else has an idea.

**Comment by simonbuehler** [comment]
Did you test with https://github.com/sensepost/objection,? Friend asked me if it would be possible to find out when the next virtual line slots would be available, haven't looked into it yet but will the next days

**Comment by simonbuehler** [comment]
https://www.kuketz-blog.de/android-tls-verifikation-und-certificate-pinning-umgehen/ worth reading
